api/classes/class-maybe-show-campaign.php

```diff
@@ -0,0 +1,167 @@
+<?php
+/**
+ * Newspack Campaigns maybe display campaign.
+ *
+ * @package Newspack
+ */
+
+/**
+ * Extend the base Lightweight_API class.
+ */
+require_once 'class-lightweight-api.php';
+
+/**
+ * GET endpoint to determine if campaign is shown or not.
+ */
+class Maybe_Show_Campaign extends Lightweight_API {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		parent::__construct();
+		$should_show = $this->should_campaign_be_shown();
+		if ( $should_show || $this->should_perform_utm_logic() ) {
+			if ( $this->should_suppress_because_utm_suppression() ) {
+				$should_show = false;
+			}
+			if ( $this->should_suppress_because_utm_medium() ) {
+				$should_show = false;
+			}
+			if ( $should_show && $this->should_suppress_because_newsletter_campaign_dismissed() ) {
+				$show_show = false;
+			}
+		}
+
+		$this->response['displayPopup'] = $should_show;
+
+		// TODO: Preview handling.
```

Comment:
Previews do not have to call the API at all – that's the approach on the exploratory [add/external-reader-service](https://github.com/Automattic/newspack-popups/tree/add/external-reader-service) branch. Just a heads up, I'll be happy to contribute to this branch :)

Reply:
Here's a quick fix - previews will use the REST API instead: https://github.com/Automattic/newspack-popups/pull/208/commits/c42783ce339efe7f3aa85947543ed50f26915bfa. The approach in `add/external-reader-service` looks good but I think we should address separately after this ships.